pkg/provisioner/ironic/ironic.go

```diff
@@ -1820,11 +1824,17 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 		// If the target state is unset while the last error is set,
 		// then the last execution of soft power off has failed.
 		if targetState == "" && ironicNode.LastError != "" {
-			return result, SoftPowerOffFailed{}
+			p.log.Info("soft power off error", "msg", ironicNode.LastError)
+			return operationFailed(ironicNode.LastError)
```

Comment:
The code in this function up to this point is now essentially common to both `softPowerOff()` and `hardPowerOff()`, so we could extract it back up into `PowerOff()`.
With that done we could allow an immediate fall back from SoftPowerOffUnsupported to trying a hard power off without reporting an error, since we'd then only call `softPowerOff()` (or `hardPowerOff()`) at the *start* of a power off attempt.